bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -98,6 +124,29 @@ public ConfiguredCallOptionsFactory(CallOptionsConfig config) {
       }
     }
 
+    @Override
+    public <RequestT> CallOptions createForRpcAttempt(RequestT request, Deadline operationDeadline) {
```

Comment:
I think it would cleaner to pass operationCallOptions instead of just the Deadline here. This way if we start putting other attributes into the CallOptions they would get preserved (see sketch below)